src/rust/scheduler/scheduler.rs

```diff
@@ -130,19 +130,20 @@ impl Scheduler {
 
     /// Insert a new task into our scheduler returning a handle corresponding to it.
     pub fn insert<F: Task>(&self, future: F) -> Option<TaskHandle> {
+        let mut task_ids: RefMut<HashMap<u64, usize>> = self.task_ids.borrow_mut();
+        // If the address space for task ids is close to half full, it will become increasingly difficult to avoid
+        // collisions, so we cap the number of tasks.
+        if task_ids.len() > MAX_NUM_TASKS {
+            panic!("Too many concurrent tasks");
+        }
+
```

Comment:
What is the motivation for this change?

Reply:
This check is done regardless of what happens in the rest of the function, so I moved the check up to fail fast and not do any other work if this condition is met.